### PR TITLE
Add new segmentation method

### PR DESCRIPTION
**Pull request by zdeveloper**
# Pull Request

## Description

1. Add new segmentation method
2. Add new hepA segmentation template created in https://github.com/CDCgov/IDWA/issues/65
3. Add new hepA labels

## Related Issues
[Link any related issues or tasks from your project management system.]

## Additional Notes
[Add any additional context or notes that reviewers should know about.]

## Checklist
Please review and complete the following checklist before submitting your pull request:

- [ ] I have ensured that the pull request is of a manageable size, allowing it to be reviewed within a single session.
- [ ] I have reviewed my changes to ensure they are clear, concise, and well-documented.
- [ ] I have updated the documentation, if applicable.
- [ ] I have added or updated test cases to cover my changes, if applicable.
- [ ] I have minimized the number of reviewers to include only those essential for the review.
- [ ] I have notified teammates in the review thread to build awareness.

## Checklist for Reviewers
Please review and complete the following checklist during the review process:

- [ ] The code follows best practices and conventions.
- [ ] The changes implement the desired functionality or fix the reported issue.
- [ ] The tests cover the new changes and pass successfully.
- [ ] Any potential edge cases or error scenarios have been considered.

### CODE REVIEW

File: OCR/ocr/services/image_segmenter.py [hunk 1]
```diff
@@ -4,8 +4,90 @@
...
+    top_left = true_points.min(axis=0)
+    # take the largest points and use them as the bottom right of your crop
+    bottom_right = true_points.max(axis=0)
+    return image[
+        top_left[0] : bottom_right[0] + 1,  # plus 1 because slice isn't
+        top_left[1] : bottom_right[1] + 1,
+    ]  # inclusive
+
+
+def segment_by_mask_then_crop(self) -> dict[str, np.ndarray]:
```
Comment (derekadombek) [inline OCR/ocr/services/image_segmenter.py:24]: Is this function being referenced anywhere yet?

Comment (zdeveloper) [reply OCR/ocr/services/image_segmenter.py:24]: good catch, that was supposed to be the default, not the segment_by_color_bounding_box. I ll make the change,

File: OCR/ocr/services/image_segmenter.py [hunk 2]
```diff
@@ -4,8 +4,91 @@
...
+    segments = {}
+
+    with open(self.labels, "r") as f:
```
Comment (arinkulshi-skylight) [inline OCR/ocr/services/image_segmenter.py:27]: can extract out the load function. 
def load_labels(self):
    try:
        with open(self.labels, "r") as f:
            return json.load(f)
    except (FileNotFoundError, json.JSONDecodeError) as e:
        raise ValueError(f"Error loading labels: {e}")

File: OCR/ocr/services/image_segmenter.py [hunk 3]
```diff
@@ -4,8 +4,91 @@
...
+        # create a mask for that color
+        mask = np.all(segmentation_template == color, axis=2).astype(int)
+
+        # add a third dimension to the mask
+        mask = mask[:, :, np.newaxis]
+
+        # multiply the original image with the mask then crop it
+        segments[label] = crop_zeros(raw_image * mask)
+
+        if self.debug is True:
```
Comment (arinkulshi-skylight) [inline OCR/ocr/services/image_segmenter.py:43]:
```
def debug_segment_info(self, label, color, mask, segments):
        print(f"label: {label}")
        print(f"color {color}")
        print("mask.shape", mask.shape)
        mask[mask >= 1] = 255  # Convert mask to binary image for saving/viewing
        cv.imwrite(f"{label}_mask.png", mask)
        if segments[label] is not None:
            print("segment.shape", segments[label].shape)
            cv.imwrite(f"{label}.png", segments[label])
        print("====")
```

**Review by arinkulshi-skylight — DISMISSED** [review]
made some minor refactoring changes. We can add them in future tickets. LGTM

**Review by arinkulshi-skylight — APPROVED** [review]
LGTM + Ruff Changes
